File: src/Foods.java
public class Foods {

    private String name;
    private int calories;
    private int fat;
    private int carbs;
    private int protein;

    public Foods(String name, int fat, int carbs, int protein) {
        // Foods per 100g
        this.name = name;
        this.fat = fat;
        this.carbs = carbs;
        this.protein = protein;
        this.calories = fat * 9 + carbs * 4 + protein * 4;
    }

    // Getters
    public String getName() {
        return name;
    }

    public int getCalories() {
        return calories;
    }

    public int getFat() {
        return fat;
    }

    public int getCarbs() {
        return carbs;
    }

    public int getProtein() {
        return protein;
    }

    // Setters
    public void setName(String name) {
        this.name = name;
    }

    public void setProtein(int protein) {
        this.protein = protein;
    }

    public void setCarbs(int carbs) {
        this.carbs = carbs;
    }

    public void setFat(int fat) {
        this.fat = fat;
    }

    public void setCalories(int calories) {
        this.calories = calories;
    }

    public void printFood() {
        System.out.println("Name: " + getName());
        System.out.println("Protein: " + getProtein() + "g");
        System.out.println("Carbs: " + getCarbs() + "g");
        System.out.println("Fat: " + getFat() + "g");
        System.out.println("Calories: " + getCalories());
    }
}

File: C++/Foods.cpp
#include "Foods.h"


Foods::Foods() : mName("None"), mProtein(0), mCarbs(0), mFat(0) {
    cout << "Foods default constructor called\n";
}

Foods::Foods(string aName, int aProtein, int aCarbs, int aFat)
        : mName(aName), mProtein(aProtein), mCarbs(aCarbs), mFat(aFat) {
    cout << "Foods 4 parameter constructor called with name: '" << mName << "'\n";
}

Foods::~Foods() {
    cout << "Foods destructor called and food with name: '" << mName << "' deleted\n";
}

string Foods::getName() const {
    return mName;
}

int Foods::getProtein() const {
    return mProtein;
}

int Foods::getCarbs() const {
    return mCarbs;
}

int Foods::getFat() const {
    return mFat;
}

int Foods::getCalories() const {
    return mCalories;
}

void Foods::setName(string aName) {
    this->mName = aName;
}

void Foods::setProtein(int aProtein) {
    this->mProtein = aProtein;
    this->setCalories();
}

void Foods::setCarbs(int aCarbs) {
    this->mCarbs = aCarbs;
    this->setCalories();
}

void Foods::setFat(int aFat) {
    this->mFat = aFat;
    this->setCalories();
}
void Foods::setCalories(){
    this->mCalories = (mProtein * 4) + (mCarbs * 4) + (mFat * 9);
}

void Foods::printFood() {
    cout << "\n-------\n";
    cout << "Name: " << getName() << endl;
    cout << "Protein: " << getProtein() << endl;
    cout << "Carbs: " << getCarbs() << endl;
    cout << "Fat: " << getFat() << endl;
    cout << "Calories: " << getCalories() << endl;
    cout << "-------\n";

}


